src/cli/commands/workspaces.js

```diff
@@ -27,8 +29,30 @@ export async function info(config: Config, reporter: Reporter, flags: Object, ar
   const publicData = {};
 
   for (const workspaceName of Object.keys(workspaces)) {
+    const {loc, manifest} = workspaces[workspaceName];
+
+    const workspaceDependencies = new Set();
+    const mismatchedWorkspaceDependencies = new Set();
+
+    for (const dependencyType of DEPENDENCY_TYPES) {
+      if (dependencyType !== 'peerDependencies') {
+        for (const dependencyName of Object.keys(manifest[dependencyType] || {})) {
+          if (Object.prototype.hasOwnProperty.call(workspaces, dependencyName)) {
```

Comment:
Don't we have a shortcut for `Object.prototype.hasOwnProperty.call` here? Do we _really_ need it?

Reply:
I don't think we have, there's multiple matches for this function in the codebase.

Reply:
Ah, we should make that happen. That said I'm not sure if we need to be so defensive about this. We should be able to use `Object.hasOwnProperty()` I guess? Or some variant of `Object.keys()` that only returns own properties?

Reply:
`Object.keys` only returns own properties, but we're calling it on a different object from the one we're dereferencing. So for example, a malicious (or a bit weird) user could add a dependency called `__proto__` to one of its workspaces, and without this check we would then access `workspaces.__proto__`.